vllm/reasoning/abs_reasoning_parsers.py

```diff
@@ -125,14 +119,16 @@ def get_reasoning_parser(cls, name) -> type:
         if name in cls.reasoning_parsers:
             return cls.reasoning_parsers[name]
 
-        raise KeyError(f"reasoning helper: '{name}' not found in "
-                       "reasoning_parsers")
+        raise KeyError(
+            f"reasoning helper: '{name}' not found in reasoning_parsers")
 
     @classmethod
-    def _register_module(cls,
-                         module: type,
-                         module_name: Optional[Union[str, list[str]]] = None,
-                         force: bool = True) -> None:
+    def _register_module(
+        cls,
+        module: type,
```

Comment:
```suggestion
        module: type[Any],
```
? or does it take in a type variable ? iirc we can also pass in a generic type here?

Reply:
This is copied from the tool parser https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/tool_parsers/abstract_tool_parser.py#L99

Should we change it too?

Reply:
OK, I created an issue to keep track. https://github.com/vllm-project/vllm/issues/15658